### PR TITLE
Expand path templates via resttemplate's uriTemplateHandler

### LINKED ISSUE #3499: RestTemplate client misses placeholdered URI's in metrics and logs a pre-build one (with replaced variables)
##### Description
Common practice of using resttemplate is to use methods that handle templated paths with variables either use `restTemplate.getUriTemplateHandler().expand()` to expand path variables, so resttemplate can store and reuse original template path while logging metrics (e.g. via `org.springframework.boot.actuate.metrics.web.client.MetricsClientHttpRequestInterceptor` in Spring Boot 2), etc.

##### openapi-generator version
master

##### Suggest a fix/enhancement
I suggest to replace static call in Api client method from:
```(java)
UriComponentsBuilder.fromPath("/pet/{petId}").buildAndExpand(uriVariables).toUriString()
```

with one that addresses uriTemplateHandler of restTemplate, eg :
```(java)
restTemplate.getUriTemplateHandler().expand("/pet/{petId}", uriVariables).toString();
```

### PR DESCRIPTION
**Pull request by unintended**
### PR checklist

- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh`, `./bin/openapi3/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`. If contributing template-only or documentation-only changes which will change sample output, be sure to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) first.
- [x] Filed the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `4.1.x`, `5.0.x`. Default: `master`.
- [x] Copied the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### Description of the PR
This pr changes uri extension from static call of UriComponentsBuilder to more common way (via RestTemplate's UriTemplateHandler) and makes possible to natively handle templated path in metric collectors, etc
fixes #3499

### CODE REVIEW

File: modules/openapi-generator/src/main/resources/Java/libraries/resttemplate/ApiClient.mustache [hunk 1]
```diff
@@ -555,6 +555,16 @@ public class ApiClient {
         return isForm ? formParams : obj;
     }
 
+    /**
+     * Expand path template with variables
+     * @param pathTempalte path template with placeholders
+     * @param variables variables to replace
+     * @return path with placeholders replaced by variables
+     */
+    public String expandPath(String pathTempalte, Map<String, Object> variables) {
```
Comment (macjohnny) [inline modules/openapi-generator/src/main/resources/Java/libraries/resttemplate/ApiClient.mustache]: Typo „pathTempalte“

**Comment by macjohnny** [comment]
please re-generate the samples

**Comment by macjohnny** [comment]
cc @bbdouglas (2017/07) @sreeshas (2017/08) @jfiala (2017/08) @lukoyanov (2017/09) @cbornet (2017/09) @jeff9finger (2018/01) @karismann (2019/03) @Zomzog (2019/04)

**Comment by wing328** [comment]
@unintended thanks for the PR, which has been included in the 4.1.0 release: https://twitter.com/oas_generator/status/1160000504455319553

**Pull request merged**